### DISCUSSION TITLE
How does the `split_bodies()` filter work in terms of labelling each split body?

### DISCUSSION THREAD

**Discussion by venugovh** [pyvista/pyvista#1638] [Q&A]
Hi,

I have an STL slice-data that contains different polygons. When I apply `split_bodies` to that slice, it splits the slice into `n` number of `UnstructuredGrid` data, each of which is a polygon. To put it in perspective, I am attaching an image of a slice. (please ignore the letter labels in the image, I did not have another image!) 

I am curious as to how Pyvista divides and orders the individual bodies? Is it based on the polygon area, bounding box size, or is there an inherent tree concept that consecutively places nested polygons one after the other?

If it does indeed have a polygon nesting-based rationale, then my life would be easy for a project I am working on currently.

If it's not, then I think it would be a really useful addition to the library. 

Looking forward to a response!

Thanks!

![image](https://user-images.githubusercontent.com/47824150/132109974-56d893be-138e-4c2d-83b1-615fa6633bc5.png)

**Comment by banesullivan** [comment]
Under the hood, this is entirely implemented with the `vtkConnectivityFilter` in the `extract all regions` mode. All ordering is determined by this filter and we simply iterate over the list in PyVista to split them.

I'm going to defer your question to the VTK docs and perhaps asking the VTK discourse: https://vtk.org/doc/nightly/html/classvtkConnectivityFilter.html#details

I suspect there is no nesting-based rationale in the VTK filter
